app/jobs/bulk_invoice_job.rb

```diff
@@ -15,8 +15,9 @@ def perform(order_ids, filepath)
 
   private
 
-  def orders_from(order_ids)
-    Spree::Order.where(id: order_ids).order("completed_at DESC")
+  # Ensures the records are returned in the same order the ids were originally given in
+  def sorted_orders(order_ids)
+    Spree::Order.where(id: order_ids).to_a.sort_by{ |order| order_ids.index(order.id) }
```

Comment:
It doesn't really matter but I think that this could be slightly more efficient:

```suggestion
    indexes = order_ids.each_with_index.to_h
    Spree::Order.where(id: order_ids).to_a.sort_by{ |order| indexes[order.id] }
```

And just for fun, not more efficient than above but maybe a clearer intention:

```suggestion
    orders_by_id = Spree::Order.where(id: order_ids).to_a.index_by(&:id)
    order_ids.map { |id| orders_by_id[id] }
```

Reply:
I don't know which one we do prefer actually ;)

Reply:
Looking at it again, I prefer the last solution because:

* it uses simple `index_by` and `map` which is easier to understand than `each_with_index.to_h`
* it avoids sorting and therefore should be the most efficient solution.